FantasmoSDK/Classes/Extensions/ARFrame+Extension.swift

```diff
@@ -7,9 +7,157 @@
 
 import ARKit
 
-extension ARFrame {
-    /// Calculate the FMPose difference of self with respect to the given frame
-    func poseWithRespectTo(_ frame: ARFrame) -> FMPose {
-        return FMPose.diffPose(self.camera.transform, withRespectTo: frame.camera.transform)
+
+public extension ARFrame {
+    
+    /// Transform of the device in the coordinate system of camera.
+    /// For camera CS the X-axis always points along the long axis of the device (from the front camera toward the Home, https://apple.co/3t1Dw33)
+    /// For device coordinate system the Y-axis always points along the long axis of the device toward the front camera - https://apple.co/2R37LJW.
+    @inline(__always) private static let transformOfDeviceInCameraCS = simd_float4x4(
```

Comment:
From what I have read: `Don’t use @inline(__always), @_inlineable, or @_transparent.` Unless you have profiled the compiled code and are trying to address a specific performance issue.

https://lists.swift.org/pipermail/swift-users/Week-of-Mon-20170227/004886.html

Reply:
Yes Lucas, here I'm trying to address the specific performance issue, though it helps not much as multiplying matrices takes 16 multiplications when using simd (neon) instructions, and multiplication in ARM is expensive operations (takes ~15 cycles). 
We don't use these inline functions much, so we are not in danger with the app size. But they are invoked often (for example when showing anchor we calculate pose of anchor 60 times a second).

I think the right way will be to see what time profiler shows and whether we get any performance gain. I will report results when ready.